tests/scripts/all.sh

```diff
@@ -5201,9 +5206,14 @@ component_test_cmake_out_of_source () {
 }
 
 component_test_cmake_as_subdirectory () {
```

Comment:
If we remove the generated files at the beginning of `test_cmake_as_subdirectory` I think we should do it in `test_cmake_as_package` and `test_cmake_as_package_install` as well, just to avoid developers wondering why it is not the same in this three sibling components.